client/database.go

```diff
@@ -17,107 +17,212 @@ package client
-		}
-		return nil, fmt.Errorf("error running transaction: %w", err)
-	}
-
-	return res, nil
-}
-
-func (db *database) CreateCollectionsFromModels(ctx context.Context, dbName string, model interface{}, models ...interface{}) error {
-	schemas, err := schema.FromCollectionModels(model, models)
+// CreateCollections creates collections in the Database using provided collection models
+func (db *Database) CreateCollections(ctx context.Context, model schema.Model, models ...schema.Model) error {
+	schemas, err := schema.FromCollectionModels(model, models...)
 	if err != nil {
 		return fmt.Errorf("error parsing model schema: %w", err)
 	}
 
-	return db.CreateCollections(ctx, dbName, schemas)
+	db.schemas = schemas
+
+	return db.createCollectionsFromSchemas(ctx, db.name, schemas)
 }
 
-func (db *database) CreateCollectionsFromDatabaseModel(ctx context.Context, dbModel interface{}) error {
+func (db *Database) createCollectionsFromDatabaseModel(ctx context.Context, dbModel schema.Model) error {
 	dbName, schemas, err := schema.FromDatabaseModel(dbModel)
 	if err != nil {
 		return err
 	}
 
-	return db.CreateCollections(ctx, dbName, schemas)
+	db.schemas = schemas
+
+	return db.createCollectionsFromSchemas(ctx, dbName, schemas)
 }
 
-func (db *database) CreateCollections(ctx context.Context, dbName string, schemas []*schema.Schema) error {
+// createCollectionsFromSchemas transactionally creates collections from the provided schema map
+func (db *Database) createCollectionsFromSchemas(ctx context.Context, dbName string, schemas map[string]*schema.Schema) error {
 	tx, err := db.driver.BeginTx(ctx, dbName)
 	if err != nil {
 		return err
 	}
 	defer func() { _ = tx.Rollback(ctx) }()
 
+	created := make(map[string]bool)
+
 	for _, v := range schemas {
 		sch, err := schema.Build(v)
 		if err != nil {
 			return err
 		}
-		log.Debug().Interface("schema", v).Str("collection", v.Name).Msg("migrateModel")
 		err = tx.CreateOrUpdateCollection(ctx, v.Name, sch)
 		if err != nil {
 			return err
 		}
+		created[v.Name] = true
+	}
+
+	if err := tx.Commit(ctx); err != nil {
+		return err
+	}
+
+	db.created = created
+
+	return nil
+}
+
+// GetCollection returns collection object corresponding to collection model T
+func GetCollection[T schema.Model](ctx context.Context, db *Database) (*Collection[T], error) {
+	var m T
+	name := schema.ModelName(m)
+	ddb := db.driver.UseDatabase(db.name)
+
+	if !db.created[name] {
+		schema, err := schema.FromCollectionModel(m)
+		if err != nil {
+			return nil, err
+		}
+		sch, err := schema.Build()
+		if err != nil {
+			return nil, err
+		}
+		if err := ddb.CreateOrUpdateCollection(ctx, name, sch); err != nil {
+			return nil, err
+		}
+		db.created[name] = true
+		db.schemas[name] = schema
 	}
 
-	return tx.Commit(ctx)
+	return &Collection[T]{name: name, driver: db.driver, db: db,
+		schema: db.schemas[name], crud: ddb}, nil
+}
+
+// Create new Database
+func (db *Database) Create(ctx context.Context) error {
```

Comment:
Maybe CreateIfNotPresent and then hide the already existing error? or also add that as an API?